internal/pkg/agent/application/coordinator/state/state.go

```diff
@@ -0,0 +1,365 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package state
```

Comment:
Is there a specific reason why this was moved to its own subpackage ?

Reply:
Yes, conversation is above.